sensing/radar_threshold_filter/src/radar_threshold_filter_node/radar_threshold_filter_node.cpp

```diff
@@ -0,0 +1,172 @@
+{
+using radar_msgs::msg::RadarReturn;
+using radar_msgs::msg::RadarScan;
+
+RadarThresholdFilterNode::RadarThresholdFilterNode(const rclcpp::NodeOptions & node_options)
+: Node("radar_threshold_filter", node_options)
+{
+  // Parameter Server
+  set_param_res_ = this->add_on_set_parameters_callback(
+    std::bind(&RadarThresholdFilterNode::onSetParam, this, _1));
+
+  // Node Parameter
+  node_param_.is_amplitude_filter =
+    declare_parameter<bool>("node_params.is_amplitude_filter", false);
+  node_param_.amplitude_min = declare_parameter<double>("node_params.amplitude_min", 0.0);
+  node_param_.amplitude_max = declare_parameter<double>("node_params.amplitude_max", 0.0);
+  node_param_.is_range_filter = declare_parameter<bool>("node_params.is_range_filter", false);
+  node_param_.range_min = declare_parameter<double>("node_params.range_min", 0.0);
+  node_param_.range_max = declare_parameter<double>("node_params.range_max", 0.0);
+  node_param_.is_azimuth_filter = declare_parameter<bool>("node_params.is_azimuth_filter", false);
+  node_param_.azimuth_min = declare_parameter<double>("node_params.azimuth_min", 0.0);
+  node_param_.azimuth_max = declare_parameter<double>("node_params.azimuth_max", 0.0);
+
+  // Subscriber
+  sub_radar_ = create_subscription<RadarScan>(
+    "~/input/radar", rclcpp::QoS{1}, std::bind(&RadarThresholdFilterNode::onData, this, _1));
+
+  // Publisher
+  pub_radar_ = create_publisher<RadarScan>("~/output/radar", 1);
+}
+
+rcl_interfaces::msg::SetParametersResult RadarThresholdFilterNode::onSetParam(
+  const std::vector<rclcpp::Parameter> & params)
+{
+  rcl_interfaces::msg::SetParametersResult result;
+  try {
+    {
+      auto & p = node_param_;
+      update_param(params, "node_params.is_amplitude_filter", p.is_amplitude_filter);
+      update_param(params, "node_params.amplitude_min", p.amplitude_min);
+      update_param(params, "node_params.amplitude_max", p.amplitude_max);
+      update_param(params, "node_params.is_range_filter", p.is_range_filter);
+      update_param(params, "node_params.range_min", p.range_min);
+      update_param(params, "node_params.range_max", p.range_max);
+      update_param(params, "node_params.is_azimuth_filter", p.is_azimuth_filter);
+      update_param(params, "node_params.azimuth_min", p.azimuth_min);
+      update_param(params, "node_params.azimuth_max", p.azimuth_max);
+    }
+  } catch (const rclcpp::exceptions::InvalidParameterTypeException & e) {
+    result.successful = false;
+    result.reason = e.what();
+    return result;
+  }
+  result.successful = true;
+  result.reason = "success";
+  return result;
+}
+
+bool RadarThresholdFilterNode::isDataReady()
+{
+  if (!radar_data_) {
+    RCLCPP_INFO_THROTTLE(get_logger(), *get_clock(), 1000, "waiting for radar msg...");
+    return false;
+  }
+  return true;
+}
+
+void RadarThresholdFilterNode::onData(const RadarScan::ConstSharedPtr msg)
+{
+  radar_data_ = msg;
+  if (!isDataReady()) {
+    return;
+  }
```

Comment:
no need?

Reply:
Fixed at [2151bd7](https://github.com/autowarefoundation/autoware.universe/pull/2300/commits/2151bd7cb8097aaeb6490cf4f2bac1dbc61bff15)

Reply:
Is the isDataReady function needed?

Reply:
Fixed at [d2f8a8b](https://github.com/autowarefoundation/autoware.universe/pull/2300/commits/d2f8a8bcaec48f559f46f0d32e8c87b111fb7913)